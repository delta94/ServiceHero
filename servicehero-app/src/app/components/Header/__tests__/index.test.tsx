import React from 'react';
import { render } from '@testing-library/react';
import { Store } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { configureAppStore } from 'store/configureStore';
import { Header } from '..';
import { BrowserRouter } from 'react-router-dom';

const renderComponent = (store: Store) =>
  render(
    <Provider store={store}>
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    </Provider>,
  );

describe('<Header  />', () => {
  let store: ReturnType<typeof configureAppStore>;

  beforeEach(() => {
    store = configureAppStore();
  });
  it('should match snapshot', () => {
    const loadingIndicator = renderComponent(store);
    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });
});
