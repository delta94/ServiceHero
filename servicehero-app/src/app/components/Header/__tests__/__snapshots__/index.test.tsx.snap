// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header  /> should match snapshot 1`] = `
.c1 {
  justify-self: right;
}

.c0 {
  display: grid;
  position: relative;
  z-index: 10;
  grid-template-columns: 1fr 1fr 1fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: rgba(255,255,255,0.5);
  font-size: 16px;
  padding: 0px 60px;
}

<header
  class="ant-layout-header c0"
>
  <div />
  <a
    class="c1"
    href="/login"
  >
    Login
  </a>
</header>
`;
